src/internal/modules/billing/controllers/two-part-tariff.js

```diff
@@ -9,32 +9,48 @@ const messages = {
   60: 'Over abstraction'
 };
 
+const getTotals = licences => {
+  const initVals = {
+    ready: 0,
+    errors: 0,
+    total: 0
+  };
+  const totalErrors = licences.reduce((acc, row) => ({
+    errors: acc.errors + (row.twoPartTariffStatuses.length > 0 ? 1 : 0)
+  }), initVals);
+
+  const totals = {
+    ...totalErrors,
+    ready: licences.length - totalErrors.errors,
+    total: licences.length
+  };
+  return totals;
+};
+
 const getErrorString = errorCodes => errorCodes.reduce((acc, code) => {
   return acc ? 'Multiple errors' : messages[code];
 }, null);
 
-const getTwoPartTariffReview = async (request, h) => {
+const getTwoPartTariffAction = async (request, h, action) => {
   const { batch } = request.pre;
   const licencesData = await services.water.billingBatches.getBatchLicences(batch.id);
-
   // gets 2pt matching error messages and define error types
   const licences = licencesData.map(licence => ({
     ...licence,
     twoPartTariffStatuses: getErrorString(licence.twoPartTariffStatuses)
   }));
 
-  // calculate total errors
-  const totals = licences.reduce((acc, row) => ({
-    errors: acc.errors + (row.twoPartTariffStatuses ? 1 : 0)
-  }), { errors: 0 });
-
-  return h.view('nunjucks/billing/two-part-tariff-review', {
+  return h.view('nunjucks/billing/two-part-tariff-' + action, {
     ...request.view,
     batch,
     licences,
-    totals,
+    totals: getTotals(licencesData),
     back: `/billing/batch/list`
   });
 };
 
+const getTwoPartTariffReview = async (request, h, action) => getTwoPartTariffAction(request, h, 'review');
+const getTwoPartTariffViewReady = async (request, h) => getTwoPartTariffAction(request, h, 'ready');
```

Comment:
There's a handy function in lodash called partialRight to create a new function by appending arguments - this would then become:
```
const getTwoPartTariffReview = partialRight(getTwoPartTariffAction, 'review');
const getTwoPartTariffViewReady = partialRight(getTwoPartTariffAction, 'ready');
```
However your implementation above may be more clear/explicit

Reply:
very handy to know - I followed a similar pattern in the bill-run controller so will probably leave it like this for now and can refactor with the new ticket i created.